src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -18,69 +18,57 @@ namespace Microsoft.ML.Runtime.Internal.CpuMath
 {
     internal static class AvxIntrinsics
     {
+        private static readonly Vector128<float> _absMask128 = Sse2.IsSupported ?
+            Sse.StaticCast<int, float>(Sse2.SetAllVector128(0x7FFFFFFF)) :
+            Sse.SetAllVector128(BitConverter.Int32BitsToSingle(0x7FFFFFFF));
+
+        private static readonly Vector256<float> _absMask256 = Avx.StaticCast<int, float>(Avx.SetAllVector256(0x7FFFFFFF));
+
         private const int Vector256Alignment = 32;
 
-        private static bool Compat(AlignedArray a)
+        private static bool HasCompatibleAlignment(AlignedArray alignedArray)
         {
-            Contracts.AssertValue(a);
-            Contracts.Assert(a.Size > 0);
-            return a.CbAlign == Vector256Alignment;
+            Contracts.AssertValue(alignedArray);
+            Contracts.Assert(alignedArray.Size > 0);
+            return (alignedArray.CbAlign % Vector256Alignment) == 0;
         }
 
-        private static unsafe float* Ptr(AlignedArray a, float* p)
+        private static unsafe float* GetAlignedBase(AlignedArray alignedArray, float* unalignedBase)
         {
-            Contracts.AssertValue(a);
-            float* q = p + a.GetBase((long)p);
-            Contracts.Assert(((long)q & (Vector256Alignment - 1)) == 0);
-            return q;
+            Contracts.AssertValue(alignedArray);
+            float* alignedBase = unalignedBase + alignedArray.GetBase((long)unalignedBase);
+            Contracts.Assert(((long)alignedBase & (Vector256Alignment - 1)) == 0);
+            return alignedBase;
         }
 
         [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
-        private static Vector256<float> ToVector256(in Vector128<float> a, in Vector128<float> b)
-        {
-            // REVIEW NEEDED: Is it the correct port of the following code?
-            // #ifndef _WIN32
-            // #define _mm256_set_m128(va, vb) _mm256_insertf128_ps(_mm256_castps128_ps256(vb), va, 1)
-            // #endif
-            return Avx.InsertVector128(Avx.ExtendToVector256(b), a, 1);
-        }
+        private static Vector256<float> SetHighLow(in Vector128<float> a, in Vector128<float> b)
+            => Avx.InsertVector128(Avx.ExtendToVector256(b), a, 1);
```

Comment:
By, this should be `SetHighLow`, I meant that there is an `Avx.SetHighLow` method, which does the right functionality 😄